yt_dlp/extractor/la7.py

```diff
@@ -25,19 +26,39 @@ class LA7IE(InfoExtractor):
         'url': 'http://www.la7.it/crozza/video/inccool8-02-10-2015-163722',
         'md5': '8b613ffc0c4bf9b9e377169fc19c214c',
         'info_dict': {
-            'id': '0_42j6wd36',
+            'id': 'inccool8-02-10-2015-163722',
             'ext': 'mp4',
             'title': 'Inc.Cool8',
             'description': 'Benvenuti nell\'incredibile mondo della INC. COOL. 8. dove “INC.” sta per “Incorporated” “COOL” sta per “fashion” ed Eight sta per il gesto atletico',
             'thumbnail': 're:^https?://.*',
-            'uploader_id': 'kdla7pillole@iltrovatore.it',
-            'timestamp': 1443814869,
             'upload_date': '20151002',
         },
     }, {
         'url': 'http://www.la7.it/omnibus/rivedila7/omnibus-news-02-07-2016-189077',
         'only_matching': True,
     }]
+    _HOST = 'https://awsvodpkg.iltrovatore.it'
+
+    def _generate_mp4_url(self, quality, m3u8_formats):
+        for f in m3u8_formats:
+            if f['vcodec'] != 'none' and quality in f['url']:
+                http_url = '%s/content/%s.mp4' % (self._HOST, quality)
+
+                urlh = self._request_webpage(
+                    HEADRequest(http_url), quality,
+                    note='Check filesize', fatal=False
+                )
+                if urlh:
+                    http_f = f.copy()
+                    del http_f['manifest_url']
+                    http_f.update({
+                        'format_id': http_f['format_id'].replace('hls-', 'https-'),
+                        'url': http_url,
+                        'protocol': 'https',
+                        'filesize_approx': int_or_none(urlh.headers.get('Content-Length', None)),
+                    })
```

Comment:
I dont know if it is really usefull to make a request just for getting the filesize. No other extractor does this

Reply:
yeah, I know, but it's maximum 2 request and since some tv programme are quite long (3 hours+) the file size can be quite big: 2GB, so to give this info to the user con be useful and, as you can see from the BEFORE output, it was something that the kaltura extractor was providing to the user.

Reply:
I am still against it, but I wont force you to remove it either. I don't use the extractor and if u think this is worth it, ig u can keep it. But if more extractors start doing this in future, I'll have to design some strict guidelines on when this is allowed and when it isnt

Reply:
ok

Reply:
what if I make it optional?

I'll create a method in the utils.py file and add on option --get-filesize (or --generate-filesize) so that the extractors developer can use the method and the user has the control to allow it or not.

Reply:
You can see my implementation in the last 2 commits

Reply:
if such a feature is to be implemented, it will need to be done in general and not specific to extractors. I am reverting your last 2 commits for this PR

There is also a feature request to fetch more format details using ffprobe. Any implementation of this will also need to atleast leave room for that